### ISSUE TITLE
Build failure due to `gi-pango`

### ISSUE THREAD

**Issue by RaitoBezarius** [phuhl/linux_notification_center#21]
Hi, thanks for the project.

I'm trying to build the project to try to send some contributions and add changes (notably translations, stuff like that).

But when I build, I have this error:

```
gi-pango                 > /tmp/stack-054c1088f1687d92/gi-pango-1.0.16/GI/Pango/Structs/AttrShape.hs:519:88: error:
gi-pango                 >     Variable not in scope:
gi-pango                 >       destroyFunc :: FunPtr GLib.Callbacks.C_DestroyNotify
gi-pango                 >     |
gi-pango                 > 519 |     result <- pango_attr_shape_new_with_data inkRect' logicalRect' data_ maybeCopyFunc destroyFunc
gi-pango                 >     |                                                                                        ^^^^^^^^^^^
gi-pango                 > 
```

I googled it and seems like it happens when the generated code is wrong, but as I am a newbie in GUI Haskell, I'm unsure how to move forward.

**Comment by RaitoBezarius** [comment]
Okay, I suppose this is due to `haskell-gi` issue on the matter and my stack won't pick up the new package for some reason. So I just brute-updated all packages and resolvers and it seems like it builds.

@phuhl Are you interested into bumping all dependencies of this project? I can spend time to fix whatever could happen and send you a PR.

**Comment by RaitoBezarius** [comment]
Well I naively tried and got this error in the project's code:
```
linux_notification_center/src/TransparentWindow.hs:148:3: error:
    • Couldn't match representation of type ‘GI.Gtk.Objects.CssProvider.CssProvider’
                               with that of ‘Data.GI.Base.BasicTypes.ManagedPtr ()’
        arising from a use of ‘styleContextAddProviderForScreen’
      The data constructor ‘GI.Gtk.Objects.CssProvider.CssProvider’
        of newtype ‘GI.Gtk.Objects.CssProvider.CssProvider’ is not in scope
    • In the expression:
        styleContextAddProviderForScreen screen provider
      In a stmt of a 'do' block:
        styleContextAddProviderForScreen screen provider
          $ fromIntegral STYLE_PROVIDER_PRIORITY_USER
      In the expression:
        do provider <- cssProviderNew
           cssProviderLoadFromData provider style
           styleContextAddProviderForScreen screen provider
             $ fromIntegral STYLE_PROVIDER_PRIORITY_USER
           return ()
```

That I'm unable to understand.
According to the docs, that's the good way to do stuff, I ensured that Gtk's version < 4.

**Comment by phuhl** [comment]
Hi @RaitoBezarius, thanks for the feedback. I am currently packed with a lot of work, because I am finishing my masters-thesis. I really appreciate that you'd like to help out and if you can, I'd happily merge PRs, but not within the next month or so, sorry. In one or two months, I might be able to figure out, what the issue with the build is, but currently, I can't sorry :(

**Comment by phuhl** [comment]
Hey there @RaitoBezarius, I found the solution. I, too, version-bumped and ran into the issue, you described. It turned out, that haskell wants to cast some types and all we needed to do, is to put into scope what the compiler needed. In this case `CssProvider`:

```
import GI.Gtk
       (styleContextRemoveClass,
-- ...
       , CssProvider(..))
```

If you are interested, take a look  at commit 10c749b116c8906ee967a6e6c69e8cb56fc026d5 as soon as I pushed it. An update will follow shortly.
